src/Security/SessionAPIAuthenticator.php

```diff
@@ -148,12 +148,13 @@ function ($userIdentifier) {
      */
     private function getUserAndUpdateSession(string $sessionId): UserInterface
     {
+        $this->authenticationService->deleteExpiredSecurityTokens();
+        $this->sessionRepository->deleteExpiredSession();
```

Comment:
this should be after isValidToken as isValidToken will check token validity and refresh it if possible (if the expired token is deleted we cannot refresh it)